Assets/Scripts/Actors/ActorComponents/ActorPhysics.cs

```diff
@@ -380,77 +395,92 @@ public void JumpCheck()
+		//	{
+		//		actor.animator.SetBool( "isJumping", true );
+		//		//actor.GetAnimator().SetBool("isSliding", false);
+		//	}
+
+		//	ChangeState( ActorStates.Jumping );
+		//}
 
 		if ( !IsInState( ActorStates.Grounded ) )
 		{
 			jumpColCheckTimer += Time.deltaTime;
 		}
-
-		lateJumpTimer += Time.deltaTime;
-		rigidbody.velocity = Vector3.ClampMagnitude( rigidbody.velocity, maxSpeed );
 	}
 
-
-
-	public void ModelControl()
+	/**
+	 * @brief Orients the model to face the direction of movement
+	 * 
+	 * @details
+	 *     Orientation is reactive to velocity, meaning the
+	 *     model will face the direction of movement, rather
+	 *     than the input direction.
+	 */
+	public void OrientSelf()
 	{
-		model.position = transform.position - modelOffset;
+		Vector3 actualVelocity = transform.position - _lastPos;
+		actualVelocity.y = 0.0f;
+
+		Vector3 intendedVelocity = rigidbody.velocity;
+		intendedVelocity.y = 0.0f;
 
-		Vector3 lookVec = inputVec;
-		lookVec.y = 0.0f;
+		Vector3 lookVec = Vector3.Lerp( actualVelocity, intendedVelocity, lookDirLerp );
 
-		if ( !WadeUtils.IsZero( lookVec ) )
+		if ( !lookVec.IsZero() )
 		{
-			model.rotation = Quaternion.Lerp( model.rotation, 
-			                                  Quaternion.LookRotation(lookVec, transform.up), 
+			transform.rotation = Quaternion.Lerp( transform.rotation,
+			                                  Quaternion.LookRotation( lookVec, transform.up ),
 			                                  Time.deltaTime * modelTurnSpeed );
 		}
+
+		_lastPos = transform.position;
+	}
+
+	protected void ChangeState( ActorStates toState )
+	{
+		currentState = toState;
+		CurrentStateMethod = stateMethodMap[currentState];
+	}
+
+	bool IsInState( ActorStates checkState )
+	{
+		return currentState == checkState;
+	}
+
+	void SetFallSpeed( float fallSpeed )
+	{
+		Vector3 moveVec = rigidbody.velocity;
+		moveVec.y = fallSpeed;
+		rigidbody.velocity = moveVec;
 	}
 
 	void OnDrawGizmosSelected()
 	{
-		// ColliderVis
-		Gizmos.color = Color.white;
-		Gizmos.DrawWireSphere( transform.position + Vector3.up * 0.5f, 0.5f );
-		Gizmos.DrawWireSphere( transform.position - Vector3.up * 0.5f, 0.5f );
-
-		// JumpCheck
-		Gizmos.color = Color.red;
-		Gizmos.DrawWireSphere( transform.position + groundPosOffset, jumpCheckRadius );
-
-		// SlopeCheck
-		Gizmos.color = Color.blue;
-		Gizmos.DrawSphere( transform.position - Vector3.up * groundSlopeCheckHeight + model.forward * 0.3f + model.right * 0.2f, groundSlopeCheckRadius );
-		Gizmos.DrawSphere( transform.position - Vector3.up * groundSlopeCheckHeight + model.forward * 0.3f - model.right * 0.2f, groundSlopeCheckRadius );
-
-		// Climb check
-		Gizmos.color = Color.green;
-		Gizmos.DrawSphere( transform.position, climbCheckRadius );
-		Gizmos.DrawSphere( transform.position + model.forward * climbCheckDistance, climbCheckRadius );
+		if ( Application.isPlaying )
+		{
+			Gizmos.DrawWireSphere( GetComponent<Transform>().position + bumper.center,
```

Comment:
Why should this only be when playing? This goes back to why we shouldn't use new private Transform. Calling GetComponent<Transform> looks totally weird.